lib/marked.js

```diff
@@ -504,7 +504,8 @@ var inline = {
   code: /^(`+)([^`]|[^`][\s\S]*?[^`])\1(?!`)/,
   br: /^ {2,}\n(?!\s*$)/,
   del: noop,
-  text: /^(`+|[^`])(?:[\s\S]*?(?:(?=[\\<!\[`*]|\b_|$)|[^ ](?= {2,}\n))|(?= {2,}\n))/
+  text: /^(`+|[^`])(?:[\s\S]*?(?:(?=[\\<!\[`*]|(?<!\w)\$|\b_|$)|[^ ](?= {2,}\n))|(?= {2,}\n))/,
```

Comment:
Unfortunately, lookbehinds aren't supported in Safari ([link](https://caniuse.com/js-regexp-lookbehind)) :/

Perhaps we could get away with changing `(?<!\w)\$` to `\B\$`? The `\B` would ensure that the character before the $ is a non-word character at least, so it wouldn't capture anything where a letter or number is before the $.

Reply:
Oh I didn't know that it was not supported, but \B does seem to work nice so I changed it!

Reply:
I was surprised to find out they worked in other browsers actually. Last I did a lot of work on this, I don't remember them being available anywhere, so that's nice to know. Hopefully Safari will catch up on that one day since it does seem clearer that way